hibernate-core/src/main/java/org/hibernate/sql/results/graph/embeddable/internal/EmbeddableForeignKeyResultImpl.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html
+ */
+package org.hibernate.sql.results.graph.embeddable.internal;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+
+import org.hibernate.engine.FetchTiming;
+import org.hibernate.metamodel.mapping.BasicValuedModelPart;
+import org.hibernate.metamodel.mapping.EmbeddableMappingType;
+import org.hibernate.metamodel.mapping.EmbeddableValuedModelPart;
+import org.hibernate.metamodel.mapping.EntityMappingType;
+import org.hibernate.metamodel.mapping.internal.SingularAssociationAttributeMapping;
+import org.hibernate.query.NavigablePath;
+import org.hibernate.sql.ast.spi.SqlSelection;
+import org.hibernate.sql.results.graph.AbstractFetchParent;
+import org.hibernate.sql.results.graph.AssemblerCreationState;
+import org.hibernate.sql.results.graph.DomainResult;
+import org.hibernate.sql.results.graph.DomainResultAssembler;
+import org.hibernate.sql.results.graph.DomainResultCreationState;
+import org.hibernate.sql.results.graph.Fetch;
+import org.hibernate.sql.results.graph.Fetchable;
+import org.hibernate.sql.results.graph.Initializer;
+import org.hibernate.sql.results.graph.basic.BasicFetch;
+import org.hibernate.sql.results.graph.basic.BasicResult;
+import org.hibernate.sql.results.graph.embeddable.EmbeddableResultGraphNode;
+import org.hibernate.sql.results.graph.entity.internal.EntityFetchSelectImpl;
+
+/**
+ * @author Andrea Boriero
+ */
+public class EmbeddableForeignKeyResultImpl<T> extends AbstractFetchParent
+		implements EmbeddableResultGraphNode, DomainResult<T> {
+
+	private final String resultVariable;
+
+	public EmbeddableForeignKeyResultImpl(
+			List<SqlSelection> sqlSelections,
+			NavigablePath navigablePath,
+			EmbeddableValuedModelPart embeddableValuedModelPart,
+			String resultVariable,
+			DomainResultCreationState creationState) {
+		super( embeddableValuedModelPart.getEmbeddableTypeDescriptor(), navigablePath );
+		this.resultVariable = resultVariable;
+		AtomicInteger atomicInteger = new AtomicInteger( 0 );
```

Comment:
We might well save some CPU cycles by using default constructor instead. Super minor.

Reply:
Wondering whether we need `AtomicInteger` here for it seems we don't need to worry about concurrency issue here. Maybe the sole reason is to have a integer wrapper which is mutable? Using `AtomicInteger` might incur unnecessary performance cost. I saw there is a much light-weight alternative in `org.hibernate.envers.internal.tools.MutableInteger`.